Mlem/Views/Shared/CollapsibleSection.swift

```diff
@@ -0,0 +1,69 @@
+//
+//  FakeSection.swift
+//  Mlem
+//
+//  Created by Sjmarf on 02/01/2024.
+//
+
+import SwiftUI
+
+struct CollapsibleSection<Content: View>: View {
+    var header: String?
+    var footer: String?
+
+    @ViewBuilder var content: () -> Content
+    @State private var collapsed: Bool
+  
+    init(
+        _ header: String? = nil,
+        footer: String? = nil,
+        collapsed: Bool = false,
+        @ViewBuilder _ content: @escaping () -> Content
+    ) {
+        self.header = header
+        self.footer = footer
+        self.content = content
+        self._collapsed = State(wrappedValue: collapsed)
+    }
+    
+    var body: some View {
+        VStack(alignment: .leading, spacing: 4) {
+            if let header {
+                HStack {
+                    Text(header)
+                        .textCase(.uppercase)
+                        .opacity(0.5)
+                    Spacer()
+                    Image(systemName: Icons.dropdown)
+                        .fontWeight(.semibold)
+                        .foregroundStyle(Color.accentColor)
+                        .rotationEffect(Angle(degrees: collapsed ? -90 : 0))
+                        .onTapGesture { withAnimation(.default) { collapsed.toggle() }}
```

Comment:
I don't love having this just on the dropdown carat--intuitively I want to be able to tap anywhere in the rectangle roughly surrounding the name all the way over to the dropdown indicator to expand/collapse, though I'm not sure how to cue that neatly. Kicking this discussion over to Slack